R/build-redirects.R

```diff
@@ -0,0 +1,45 @@
+build_redirects <- function(pkg = ".",
+                            override = list()) {
+  pkg <- section_init(pkg, depth = 1L, override = override)
+
+  if (is.null(pkg$meta$redirects)) {
+    return(invisible())
+  }
+
+  rule("Building redirects")
+  if (is.null(pkg$meta$url)) {
+    abort(sprintf("Can't find %s.", pkgdown_field(pkg, "url")))
+  }
+
+  sitemap <- xml2::read_xml(file.path(pkg$dst_path, "sitemap.xml"))
+  paths <- xml2::xml_contents(sitemap) %>%
+    purrr::map_chr(xml2::xml_text) %>%
+    get_url_paths()
+
+  purrr::walk(
+    pkg$meta$redirects,
+    build_redirect,
+    pkg = pkg,
+    paths = paths
+  )
+}
+
+build_redirect <- function(entry, pkg, paths) {
+  new <- entry[2]
```

Comment:
Maybe check that `entry` is a length two character vector, and error if not? (Using `iwalk` instead of `walk` to make a better errror)